framework/src/utils/MatrixTools.C

```diff
@@ -43,7 +45,14 @@ void
 inverse(std::vector<PetscScalar> & A, unsigned int n)
 {
   mooseAssert(n >= 1, "MatrixTools::inverse - n (leading dimension) needs to be positive");
-  mooseAssert(n == static_cast<int>(n), "MatrixTools::inverse - n (leading dimension) too large");
+
+#ifndef NDEBG
+  constexpr auto N = CHAR_BIT * sizeof(unsigned int);
+
+  std::bitset<N> most_significant_bit_test(n);
```

Comment:
But the max signed value can be safely cast to unsigned anyways.

On Tue, Jun 14, 2016, 7:16 AM Cody Permann <notifications@github.com> wrote:

> In framework/src/utils/MatrixTools.C
> <https://github.com/idaholab/moose/pull/7183#discussion_r66978275>:
>
> > @@ -43,7 +45,14 @@ void
> >  inverse(std::vector<PetscScalar> & A, unsigned int n)
> >  {
> >    mooseAssert(n >= 1, "MatrixTools::inverse - n (leading dimension) needs to be positive");
> > -  mooseAssert(n == static_cast<int>(n), "MatrixTools::inverse - n (leading dimension) too large");
> > +
> > +#ifndef NDEBG
> > +  constexpr auto N = CHAR_BIT * sizeof(unsigned int);
> > +
> > +  std::bitset<N> most_significant_bit_test(n);
>
> Honestly I didn't try. I thought that there'd still be a type comparison
> problem.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7183/files/3d359ba5b9ba3b80b951b352e9d2cd9f7a20a6b3#r66978275>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMWPvgNAKVlkRhXy4WKQVAXXM32upSkks5qLre7gaJpZM4I0vZG>
> .
>